### PR TITLE
doc: 17.2.8 Release Notes

### PR DESCRIPTION
**Pull request by yuriw**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/releases/quincy.rst
Comment (anthonyeleven) [inline doc/releases/quincy.rst:20]: no apostrophe

File: doc/releases/quincy.rst
Comment (anthonyeleven) [inline doc/releases/quincy.rst:23]: with the

File: doc/releases/quincy.rst
Comment (anthonyeleven) [inline doc/releases/quincy.rst:24]: locally on the cluster, or on the client?

File: doc/releases/quincy.rst
Comment (anthonyeleven) [inline doc/releases/quincy.rst:27]: to the

File: doc/releases/quincy.rst
Comment (anthonyeleven) [inline doc/releases/quincy.rst:30]: via the

File: doc/releases/quincy.rst
Comment (anthonyeleven) [inline doc/releases/quincy.rst:32]: via the

File: doc/releases/quincy.rst
Comment (anthonyeleven) [inline doc/releases/quincy.rst:33]: to the

**Review by anthonyeleven — COMMENTED** [review]
Suggestions left.  Also note the Sphinx errors, which should be straightforward to resolve.

File: doc/releases/quincy.rst
Comment (idryomov) [inline doc/releases/quincy.rst:13]: I don't think we are planning another quincy release -- should this be "... eighth, and expected to be last, backport ..."?

**Comment by anthonyeleven** [comment]
> > Suggestions left. Also note the Sphinx errors, which should be straightforward to resolve.
> 
> @anthonyeleven would you like to push your corrections? TIA

My git-fu is weak, but I can do so in a followup PR after merge.

File: doc/releases/quincy.rst
Comment (yuriw) [reply doc/releases/quincy.rst:24]: don't know what this is :(

**Comment by yuriw** [comment]
> > > Suggestions left. Also note the Sphinx errors, which should be straightforward to resolve.
> > 
> > 
> > @anthonyeleven would you like to push your corrections? TIA
> 
> My git-fu is weak, but I can do so in a followup PR after merge. I did tick off the checklist.

I belive I fixed almost all of them

**Comment by anthonyeleven** [comment]
lgtm, defer to others for mergability

**Review by anthonyeleven — APPROVED** [review]

**Comment by yuriw** [comment]
jenkins test this please

**Comment by yuriw** [comment]
jenkins test docs

File: doc/releases/quincy.rst
Comment (idryomov) [reply doc/releases/quincy.rst:24]: On the client -- this says that if certain conditions are met, the client wouldn't need to send any OSD requests to service a diff-iterate request.

File: doc/releases/quincy.rst
Comment (ljflores) [inline doc/releases/quincy.rst:224]:
```suggestion
- doc/cephfs/client-auth<span></span>.rst: correct fs authorize cephfs1 /dir1 clie… ([pr#55247](https://github.com/ceph/ceph/pull/55247), 叶海丰)
```

File: doc/releases/quincy.rst
Comment (ljflores) [inline doc/releases/quincy.rst:828]:
```suggestion
- mds: disable `defer\_client\_eviction\_on\_laggy\_osds` by default ([issue#64685](http://tracker.ceph.com/issues/64685), [pr#56195](https://github.com/ceph/ceph/pull/56195), Venky Shankar)
```

**Review by ljflores — COMMENTED** [review]
I added some corrections for a few formatting warnings.

**Review by ljflores — APPROVED** [review]

**Comment by yuriw** [comment]
@anthonyeleven I suspect your commit was not signed off, can you pls fix it?

**Comment by anthonyeleven** [comment]
? I didn't make a commit AFAIK

**Comment by cbodley** [comment]
jenkins test signed

**Comment by yuriw** [comment]
jenkins test signed

File: doc/releases/quincy.rst
Comment (neha-ojha) [inline doc/releases/quincy.rst:15]: Suggestion to add a note here:

***
Users should expect to see the el8 rpm subdirectory empty and dnf commands are expected to fail with 17.2.8. They can choose to use 17.2.8 RPM packages for centos 8/el8 provided by CERN as a community member or continue to stay at 17.2.7 following instructions from https://docs.ceph.com/en/latest/install/get-packages/#rhel, the ceph.repo file should point to https://download.ceph.com/rpm-17.2.7/el8 instead of https://download.ceph.com/rpm-quincy/el8

These CERN packages come with no warranty and have not been tested. The software in them has been
tested by Ceph according to [platforms](https://docs.ceph.com/en/latest/start/os-recommendations/#platforms).
The repository for el8 builds is hosted by CERN on [Linux @ CERN](https://linuxsoft.cern.ch/repos/ceph-ext-quincy8el-stable/).
The public part of the GPG key used to sign the packages is available at 
[RPM-GPG-KEY-Ceph-Community](https://linuxsoft.cern.ch/repos/RPM-GPG-KEY-Ceph-Community). 
***

**Comment by yuriw** [comment]
jenkins test signed

**Comment by yuriw** [comment]
jenkins test docs

File: doc/releases/quincy.rst
Comment (yuriw) [reply doc/releases/quincy.rst:37]: @idryomov what do you want to change in the paragraph above?
Or do you at all?

Comment (idryomov) [reply doc/releases/quincy.rst:37]: I'm fine with it as is -- it's already out there in pacific, reef and squid release notes.

**Review by cbodley — APPROVED** [review]

File: doc/releases/quincy.rst
Comment (idryomov) [inline doc/releases/quincy.rst:51]: The changelog is full of `<span>`, `</span>` and `\` (escape) noise in various combinations, which is there on the rendered page as well.  Also, we have never had blank lines between changelog entries before.

Was there a change made to the changlog generation script?

File: doc/releases/quincy.rst
Comment (idryomov) [inline doc/releases/quincy.rst:20]: This doesn't say why or provide context.  It would be nice to mention that RPM packages for el9 are available instead.

File: doc/releases/quincy.rst
Comment (yuriw) [reply doc/releases/quincy.rst:51]: I don't know, but added a tracker for it https://tracker.ceph.com/issues/68987
@neha-ojha @ljflores ?

Comment (yuriw) [reply doc/releases/quincy.rst]: @idryomov it was a mistake and I belive it's fixed now, pls take a look

**Comment by yuriw** [comment]
@anthonyeleven @ljflores I had to replace the body of the "change log" with all PRs list, please review
TIA

**Comment by anthonyeleven** [comment]
Several instances where CentOS caps aren't ideal, but lgtm

**Comment by anthonyeleven** [comment]
jenkins retest this please

**Comment by yuriw** [comment]
jenkins test signed

**Comment by yuriw** [comment]
jenkins test docs

**Comment by yuriw** [comment]
> Several instances where CentOS caps aren't ideal, but lgtm. We'll need to get past the repository error with the Docs build though.

docs still failing
who can fix it? @anthonyeleven

**Comment by yuriw** [comment]
> > Several instances where CentOS caps aren't ideal, but lgtm. We'll need to get past the repository error with the Docs build though.
> 
> docs still failing who can fix it? @anthonyeleven

OK fixed it by finding a PR with `''` in the title and removing it

**Comment by anthonyeleven** [comment]
Groovy.  I’m traveling right now without a laptopOn Nov 21, 2024, at 12:45 PM, Yuri Weinstein ***@***.***> wrote:﻿


Several instances where CentOS caps aren't ideal, but lgtm. We'll need to get past the repository error with the Docs build though.

docs still failing who can fix it? @anthonyeleven

OK fixed it by finding a PR with '' in the title and removing it

File: doc/releases/quincy.rst
Comment (idryomov) [inline doc/releases/quincy.rst:40]: Mentioning 17.2.8 on its own line again seems redundant:
```suggestion
```

Comment (idryomov) [reply doc/releases/quincy.rst:40]: Instead, I'd move `Notable Changes` heading here, leaving the RPM announcement which is not a bulleted list but rather just paragraphed text above it.  This is how a similar announcement was formatted for 18.2.4 release.

File: doc/releases/quincy.rst
Comment (idryomov) [inline doc/releases/quincy.rst:18]: I suggested for `Notable Changes` heading to be moved, not for the second (duplicate) one to be added.  Remove the first one:
```suggestion
```

**Review by idryomov — APPROVED** [review]
Ack from the RBD perspective

**Comment by idryomov** [comment]
However, it's pretty clear that a good number of release notes are missing.   For example

> `ceph config dump --format <json|xml>` output will display the localized option names instead of its normalized version. For e.g., "mgr/prometheus/x/server_port" will be displayed instead of "mgr/prometheus/server_port". This matches the output of the non pretty-print formatted version of the command.

from https://github.com/ceph/ceph/pull/53886 and

> RADOS: A detailed version of the `balancer status` CLI command in the balancer module is now available. Users may run `ceph balancer status detail` to see more details about which PGs were updated in the balancer's last optimization.  See https://docs.ceph.com/en/latest/rados/operations/balancer/ for more information.

from https://github.com/ceph/ceph/pull/55264 -- I stopped after finding two PRs.  This is caused by backport PRs often misapplying release notes (adding them to the wrong section of `PendingReleaseNotes` along with stuff that isn't even getting backported).  I'd encourage component leads to review `Notable Changes` list and manually pick out release notes from the mess that  quincy's `PendingReleaseNotes` has become.

**Pull request merged**
